app/src/main/java/org/p2p/wallet/history/ui/history/HistoryFragment.kt

```diff
@@ -36,88 +39,101 @@ import org.p2p.wallet.utils.viewbinding.viewBinding
 import org.p2p.wallet.utils.withArgs
 import kotlin.math.absoluteValue
 
+private const val EXTRA_TOKEN = "EXTRA_TOKEN"
+
 class HistoryFragment :
     BaseMvpFragment<HistoryContract.View, HistoryContract.Presenter>(R.layout.fragment_history),
     HistoryContract.View {
 
     companion object {
-        private const val EXTRA_TOKEN = "EXTRA_TOKEN"
-        fun create(token: Token.Active): HistoryFragment = HistoryFragment().withArgs(
-            EXTRA_TOKEN to token
-        )
+        fun create(tokenForHistory: Token.Active): HistoryFragment =
+            HistoryFragment()
+                .withArgs(EXTRA_TOKEN to tokenForHistory)
     }
 
-    override val presenter: HistoryContract.Presenter by inject { parametersOf(token) }
+    override val presenter: HistoryContract.Presenter by inject { parametersOf(tokenForHistory) }
 
-    private val token: Token.Active by args(EXTRA_TOKEN)
+    private val tokenForHistory: Token.Active by args(EXTRA_TOKEN)
 
     private val historyAdapter: HistoryAdapter by unsafeLazy {
         HistoryAdapter(
-            onTransactionClicked = { presenter.onItemClicked(it) },
-            onRetryClicked = { presenter.fetchNextPage() }
+            onTransactionClicked = presenter::onItemClicked,
+            onRetryClicked = presenter::fetchNextPage
```

Comment:
If you look into the bytecode, these type of calls use reflection. 
It's okay to use it in some cases but not in the whole app

Reply:
hmm, really? i've never known. i'll fix it ofc

Reply:
but no reflection found here...
what can you say in your defence??


<img width="694" alt="Screen Shot 2022-03-25 at 9 38 48 pm" src="https://user-images.githubusercontent.com/100791035/160181787-df68ae18-0a33-47ff-9527-4c340a0ca6f1.png">

Reply:
Couldn't find on my side. Let's merge it. Will come back after workaround